### DISCUSSION TITLE
Filter recent files in Attachment Picker

### DISCUSSION THREAD

**Discussion by rupinderjeet** [GetStream/stream-chat-android#2489] [Q&A]
Is there a way to filter the recent files in Attachment Picker by name or type?

**Comment by filbabic** [comment]
Hey @rupinderjeet!

In Compose, you could format the files and build custom UI that allows for searching, but so far we don't have this behavior by default.

**Reply by rupinderjeet** [reply]
Hello @filbabic 

I wanted to filter the files before they are shown to user, not search or filter them after they are shown. 

For e.g., I don't want to support uploads for PDF files, and I disallow them when a user tries to upload them. But, the attachment picker shows PDF files as a supported format for user to tap and upload. 

Maybe I could set a setting somewhere to always disallow suggesting PDF files (perhaps using a predicate).

I am no longer working with streamchat, so, I may not be fully aware if you guys have added a support for this now. If your reply was really about this, I am sorry I misunderstood you.

Thank you for the reply.

**Comment by filbabic** [comment]
Hey @rupinderjeet That's a good piece of feedback - yes, we're looking into disallowing this.

We have #2166 #2876 tickets that'll connect to the API and read which file/mime types are supported. As soon as that's in, we'll be able to preemptively filter out these file types from the picker.

It's aimed to be focused on one of the next sprints.

**Reply by rupinderjeet** [reply]
Thank you for the update.
